backend/controller/cronjobs/cronjobs.go

```diff
@@ -0,0 +1,483 @@
+	svc.scheduler.Singleton(backoff.Backoff{Min: time.Second, Max: time.Minute}, svc.killOldJobs)
+
+	go svc.watchForUpdates(ctx)
+
+	return svc
+}
+
+func (s *Service) NewCronJobsForModule(ctx context.Context, module *schemapb.Module) (jobs []dal.CronJob, err error) {
+	logger := log.FromContext(ctx)
+
+	start := s.clock.Now().UTC()
+	newJobs := []dal.CronJob{}
+	for _, decl := range module.Decls {
+		if verb, ok := decl.Value.(*schemapb.Decl_Verb); ok {
+			for _, metadata := range verb.Verb.Metadata {
+				if cronMetadata, ok := metadata.Value.(*schemapb.Metadata_CronJob); ok {
+					cronStr := cronMetadata.CronJob.Cron
+					schedule, err := cron.Parse(cronStr)
+					if err != nil {
+						logger.Errorf(err, "failed to parse cron schedule %q", cronStr)
+						continue
+					}
+					next, err := cron.NextAfter(schedule, start, false)
+					if err != nil {
+						logger.Errorf(err, "failed to calculate next execution for cron job %v:%v with schedule %q", module.Name, verb.Verb.Name, schedule)
+						continue
+					}
+					newJobs = append(newJobs, dal.CronJob{
+						Key:           model.NewCronJobKey(module.Name, verb.Verb.Name),
+						Ref:           schema.Ref{Module: module.Name, Name: verb.Verb.Name},
+						Schedule:      cronStr,
+						StartTime:     start,
+						NextExecution: next,
+						State:         dal.JobStateIdle,
+						// DeploymentKey: Filled in by DAL
+					})
+				}
+			}
+		}
+	}
+	return newJobs, nil
+}
+
+func (s *Service) CreatedOrReplacedDeloyment(ctx context.Context, newDeploymentKey model.DeploymentKey) {
+	// Rather than finding old/new cron jobs and updating our state, we can just reset the list of jobs
+	_ = s.resetJobsWithNewDeploymentKey(ctx, optional.Some(newDeploymentKey))
+}
+
+// resetJobs is run periodically via a scheduled task
+func (s *Service) resetJobs(ctx context.Context) (time.Duration, error) {
+	err := s.resetJobsWithNewDeploymentKey(ctx, optional.None[model.DeploymentKey]())
+	if err != nil {
+		return 0, err
+	}
+	return jobResetInterval, nil
+}
+
+// resetJobsWithNewDeploymentKey resets the list of jobs and marks the deployment key as added so that it can overrule the hash ring for a short time.
+func (s *Service) resetJobsWithNewDeploymentKey(ctx context.Context, deploymentKey optional.Option[model.DeploymentKey]) error {
+	logger := log.FromContext(ctx)
+
+	jobs, err := s.dal.GetCronJobs(ctx)
+	if err != nil {
+		logger.Errorf(err, "failed to get cron jobs")
+		return fmt.Errorf("%s: %w", "failed to get cron jobs", err)
+	}
+	s.jobChanges.Publish(jobChange{
+		changeType:         resetJobs,
+		jobs:               jobs,
+		addedDeploymentKey: deploymentKey,
+	})
+	return nil
+}
+
+func (s *Service) executeJob(ctx context.Context, job dal.CronJob) {
+	logger := log.FromContext(ctx)
+	requestBody := map[string]any{}
+	requestJSON, err := json.Marshal(requestBody)
+	if err != nil {
+		logger.Errorf(err, "could not build body for cron job: %v", job.Key)
```

Comment:
When you get around to recording events for cron, all of these error logs should be recorded and associated with the cron event.